File: autopnp_scenario/scripts/Next_Room_client.py
#! /usr/bin/env python

import roslib; roslib.load_manifest('autopnp_scenario')
import rospy
import smach
import smach_ros
import cv
import numpy as np

import actionlib
from nav_msgs.msg import OccupancyGrid 

from sensor_msgs.msg import Image
from cv_bridge import CvBridge, CvBridgeError

import autopnp_scenario.msg


    
def NextRoom( InputImg, RoomCenterX, RoomCenterY , mapResolution , mapOriginX , mapOriginY ):  
    
    client = actionlib.SimpleActionClient( 'Find_Next_Room', autopnp_scenario.msg.Next_RoomAction )

    rospy.loginfo("Waiting for the Next Room Action server to start.")

    client.wait_for_server()    

    rospy.loginfo("Next Room Action server started, sending goal.")    
      
    goal = autopnp_scenario.msg.Next_RoomGoal( input_img = InputImg, 
                                               room_center_x = RoomCenterX, 
                                               room_center_y = RoomCenterY,
                                               map_resolution = mapResolution,
                                               Map_Origin_x = mapOriginX,
                                               Map_Origin_y = mapOriginY )
    
    client.send_goal(goal)
    
    finished_before_timeout = client.wait_for_result()
    if finished_before_timeout:
        state = client.get_state()
        if state is 3:
            state = 'SUCCEEDED'
            rospy.loginfo("Action finished: %s " % state)
        else:
            rospy.loginfo("Action finished: %s " % state)

    else:
        rospy.loginfo("Action did not finish before the time out.")
        
    return client.get_result()     

File: autopnp_scenario/ros/src/next_room_client.cpp
#include <iostream>

#include <ros/ros.h>

#include <cv_bridge/cv_bridge.h>
#include <opencv/cv.h>
#include <opencv/highgui.h>

#include <actionlib/client/simple_action_client.h>
#include <actionlib/client/terminal_state.h>

#include <autopnp_scenario/NextRoomAction.h>

int main (int argc, char **argv)
	{
		ros::init(argc, argv, "test_Find_Next_Room");

		actionlib::SimpleActionClient<autopnp_scenario::NextRoomAction> ac("Find_Next_Room", true);

		ROS_INFO("Waiting for action server to start.");

		// wait for the action server to start
		ac.waitForServer(); //will wait for infinite time

		ROS_INFO("Action server started, sending goal.");
		// send a goal to the action

		autopnp_scenario::NextRoomActionGoal goal;

		//goal.map_resolution =
		//goal.Map_Origin_x =
		//goal.Map_Origin_y =

		//goal.input_img =
		//goal.room_center_x =
		//goal.room_center_y =

		//ac.sendGoal(goal);

		//wait for the action to return
		bool finished_before_timeout = ac.waitForResult(ros::Duration(30.0));

		if (finished_before_timeout)
			{
				actionlib::SimpleClientGoalState state = ac.getState();
				ROS_INFO("Action finished: %s",state.toString().c_str());
			}

		else
		ROS_INFO("Action did not finish before the time out.");

		//exit
		return 0;
	}
